### PR TITLE
Fix race condition in buffer handler

### PR DESCRIPTION
**Pull request by alexrsagen**
On very rare occasions, this line would be called after a cleanup:

```javascript
this.nativeStream.code(chunk, !this.synchronous);
```

A cleanup would set `this.nativeStream` to `null`, thereby causing an error.

Inside `this.nativeStream.bufferHandler` an `error-cleanup` would be emitted whenever it got passed an error, causing a cleanup, but the code would continue to eventually run the line above because of a missing return.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "lzma-native",
-  "version": "3.0.0",
+  "version": "3.0.1",
```
Comment (addaleax) [inline package.json]: This is going to fail CI if it doesn’t match the version in `index.js` :/

Comment (alexrsagen) [inline package.json]: Funny, never seen anything like that. Fixed in ae9ec91.

**Comment by alexrsagen** [comment]
@addaleax I fixed the file modes and bumped the version in index.js as well now.

**Comment by addaleax** [comment]
Thanks, merged as 8d7575703160b9a789f264bdd722871e589cde24 and released in `lzma-native@3.0.1`!

**Pull request closed without merging**